File: content/examples/other/leetcode/problems/product-of-array-except-self/example-1/solution.js
function productExceptSelf(nums) {
  const out = [];
  for (let i=0;i<nums.length;i+=1) {
    let prod = 1;
    for (let j=0;j<nums.length;j+=1) {
      if (i===j) continue;
      prod *= nums[j];
    }
    out.push(prod);
  }
  return out;
}

if (require.main===module) {
  console.log(productExceptSelf([1,2,3,4]));
}

module.exports = { productExceptSelf };

File: content/examples/other/leetcode/problems/product-of-array-except-self/example-1/Solution.java
import java.util.*;

public class Solution {
  public static int[] productExceptSelf(int[] nums) {
    int n = nums.length;
    int[] out = new int[n];
    for (int i = 0; i < n; i++) {
      int prod = 1;
      for (int j = 0; j < n; j++) {
        if (i == j) continue;
        prod *= nums[j];
      }
      out[i] = prod;
    }
    return out;
  }

  public static void main(String[] args) {
    System.out.println(Arrays.toString(productExceptSelf(new int[]{1,2,3,4})));
  }
}
